src/uu/mktemp/src/mktemp.rs

```diff
@@ -150,7 +150,26 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
             (template, tmp)
         }
     } else {
-        (template, PathBuf::from(tmpdir))
+        // In this case, the command was `mktemp --tmpdir=foo XXX`, so
+        // we need to parse out the parent directory and the filename
+        // from the argument `XXX`, since it may be include path
+        // separators.
+
+        // The template cannot be an absolute path. For example, `mktemp
+        // --tmpdir=a /XXX` is not allowed.
+        if path.is_absolute() {
+            return Err(MkTempError::InvalidTemplate(template.into()).into());
+        }
+
+        // `mktemp --tmpdir=a b/cXXX` becomes `a`, `b`, and `cXXX`.
+        // Then the `tmpdir` is `a/b` and `template` is `cXXX`.
+        let subdirectory = path.parent().unwrap_or_else(|| Path::new(""));
+        let tmpdir = PathBuf::from(tmpdir).join(subdirectory);
+        let template = match path.file_name() {
+            None => template,
```

Comment:
Is this ever reached? It seems to me that turning the template into a path and getting the file name of that always succeeds. If it's never reached an `expect` call might be more appropriate.

Reply:
According to https://doc.rust-lang.org/std/path/struct.Path.html#method.file_name, "Returns `None` if the path terminates in `..`.", as in
```rust
assert_eq!(None, Path::new("foo.txt/..").file_name());
```
so I think this line can be reached if we run `mktemp XXX/..`. There is some code later on that checks for a `/` in the suffix of the template and returns an error if it finds one. (The code for dealing with the prefix, randomized part, and suffix is rather complicated and distributed right now, I hope to streamline it in a future pull request.) I can add a test case for that:
```sh
$ mktemp XXX/..
mktemp: invalid suffix ‘/..’, contains directory separator
```

Reply:
All right, makes sense, you can leave it like this then